app/uk/gov/hmrc/exports/models/declaration/EoriSource.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.models.declaration
+
+import play.api.libs.json.{JsString, JsonValidationError, Reads, Writes}
+import uk.gov.hmrc.exports.models.Eori
+
+sealed trait EoriSource
+
+object EoriSource {
+  case object UserEori extends EoriSource
+  case object OtherEori extends EoriSource
+
+  val values = Seq(UserEori, OtherEori)
+
+  lazy val lookupByValue: Map[String, EoriSource] = values.map(entry => entry.toString -> entry).toMap
```

Comment:
Nit: does it really need to be `lazy`? The implicit `reads` is not, so maybe also `lookupByValue` could not be `lazy`, isn't it?